docs/advanced_examples/KNearestNeighbors.ipynb

```diff
@@ -0,0 +1,431 @@
+    "    n_samples=20, n_features=20, n_informative=3, n_redundant=0, n_classes=2, n_clusters_per_class=1\n",
+    ")\n",
+    "# Split the data-set into a train and testing sets\n",
+    "X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.5, random_state=42)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Model instantiation"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# The novel aspect introduced by Concret-ML models is the hyperparameter `n_bits`, which represents\n",
+    "# the precision for quantizing input data\n",
+    "# This quantization step is essential after the training phase, since FHE exclusively operates\n",
+    "# over integers\n",
+    "\n",
+    "n_neighbors = 3\n",
+    "\n",
+    "concrete_knn = ConcreteKNeighborsClassifier(n_bits=3, n_neighbors=n_neighbors)\n",
+    "\n",
+    "# Fit both the Concrete ML and its equivalent float estimators on clear data\n",
+    "concrete_knn, sklearn_model = concrete_knn.fit_benchmark(X_train, y_train)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Compile the model"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Saved in 47\n",
+      "Compilation time: 2.56 seconds\n"
+     ]
+    }
+   ],
+   "source": [
+    "# The compilation step aims to:\n",
+    "# - convert the quantized model to its FHE equivalent\n",
+    "# - create an executable operation graph\n",
+    "# - check the operation graph's compatibility with FHE\n",
+    "# - compute the maximum bit-width needed for model execution\n",
+    "# - determine cryptographic parameters necessary for generating secret keys and evaluation keys\n",
+    "time_begin = time.time()\n",
+    "circuit = concrete_knn.compile(X)\n",
+    "print(f\"Compilation time: {time.time() - time_begin:.2f} seconds\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Generating a key for an 8-bit circuit\n"
+     ]
+    }
+   ],
+   "source": [
+    "print(f\"Generating a key for an {circuit.graph.maximum_integer_bit_width()}-bit circuit\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Key generation"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Key generation time: 829.81 seconds\n"
+     ]
+    }
+   ],
+   "source": [
+    "# Note that this step may be time-consuming for circuits exceeding 8-bits\n",
+    "time_begin = time.time()\n",
+    "circuit.client.keygen(force=False)\n",
```

Comment:
not sure we want to expose the `force=False` feature imo

Reply:
removed